tests/pds/pds_resiliency_test.go

```diff
@@ -1197,3 +1197,148 @@ var _ = Describe("{RestoreDSDuringPXPoolExpansion}", func() {
+		bkpTarget, err = bkpClient.CreateAwsS3BackupCredsAndTarget(tenantID, fmt.Sprintf("%v-aws", bkpTargetName), deploymentTargetID)
+		log.FailOnError(err, "Failed to create S3 backup target.")
+		log.InfoD("AWS S3 target - %v created successfully", bkpTarget.GetName())
+		awsBkpTargets = append(awsBkpTargets, bkpTarget)
+		//Initializing the parameters required for workload generation
+		wkloadParams = pdsdriver.LoadGenParams{
+			LoadGenDepName: params.LoadGen.LoadGenDepName,
+			Namespace:      params.InfraToTest.Namespace,
+			NumOfRows:      params.LoadGen.NumOfRows,
+			Timeout:        params.LoadGen.Timeout,
+			Replicas:       params.LoadGen.Replicas,
+			TableName:      params.LoadGen.TableName,
+			Iterations:     params.LoadGen.Iterations,
+			FailOnError:    params.LoadGen.FailOnError,
+		}
+	})
+	It("Deploy Dataservices and Restore during KVDB failover", func() {
+		var deployments = make(map[PDSDataService]*pds.ModelsDeployment)
+		var depList []*pds.ModelsDeployment
+		Step("Deploy Data Services", func() {
+			for _, ds := range params.DataServiceToTest {
+				Step("Deploy and validate data service", func() {
+					isDeploymentsDeleted = false
+					deployment, _, _, err = DeployandValidateDataServices(ds, params.InfraToTest.Namespace, tenantID, projectID)
+					log.FailOnError(err, "Error while deploying data services")
+					deployments[ds] = deployment
+					depList = append(depList, deployment)
+					deploymentsToBeCleaned = append(deploymentsToBeCleaned, deployment)
+					deps = append(deps, deployment)
+					log.InfoD("Number of backups to be taken are- %v", len(deps))
+					dsEntity = restoreBkp.DSEntity{
+						Deployment: deployment,
+					}
+				})
+			}
+		})
+		Step("Running Workloads before taking backups", func() {
+			for _, pdsDeployment := range deps {
+				ckSum, wlDep, err := dsTest.InsertDataAndReturnChecksum(pdsDeployment, wkloadParams)
+				wlDeploymentsToBeCleaned = append(wlDeploymentsToBeCleaned, wlDep)
+				log.FailOnError(err, "Error while Running workloads")
+				log.Debugf("Checksum for the deployment %s is %s", *pdsDeployment.ClusterResourceName, ckSum)
+				pdsdeploymentsmd5Hash[*pdsDeployment.ClusterResourceName] = ckSum
+			}
+		})
+		Step("Perform multiple adhoc backup and validate them", func() {
+			log.Infof("Deployment ID: %v, backup target ID: %v", deployment.GetId(), bkpTarget.GetId())
+			for _, pdsDeployment := range deps {
+				log.Infof("Deployment ID: %v, backup target ID: %v", pdsDeployment.GetId(), bkpTarget.GetId())
+				err = bkpClient.TriggerAndValidateAdhocBackup(pdsDeployment.GetId(), bkpTarget.GetId(), "s3")
+				log.FailOnError(err, "Failed while performing adhoc backup")
+			}
+
+		})
+		Step("Trigger restore during KVDB pods are down", func() {
+			for _, deployment := range deployments {
+				failuretype := pdslib.TypeOfFailure{
+					Type: RestoreDSDuringKVDBFailOver,
+					Method: func() error {
+						return KillKvdbMasterNodeAndFailover()
+					},
```

Comment:
this will kill master node once or do you want to try killing all kvdb nodes?

Reply:
we want to try delete all while restore is running